### ISSUE TITLE
Object function Object() { [native code] } has no method 'assign'

### ISSUE THREAD

**Issue by got3nks** [christkv/mongodb-core#109]
```
node_modules/mongodb/node_modules/mongodb-core/lib/topologies/server.js:320
  self.s.pool = new Pool(Object.assign(self.s.options, options, {bson: this.s.
                                ^
TypeError: Object function Object() { [native code] } has no method 'assign'
```

**Comment by dachev** [comment]
+1

**Comment by yanCode** [comment]
at this very moment, this bug is still there....

**Comment by benurb** [comment]
Seems to be fixed with 8513f29edb0348118890b67a82a7720cc4f0fc3b, but is not yet released/published

**Comment by christkv** [comment]
fixed in 2.0.4 and mongodb driver 2.2.1
